### PR TITLE
'install' refactoring 1

### PR DESCRIPTION
**Pull request by a-ok123**

### CODE REVIEW

File: cmd/install.go [hunk 1]
```diff
@@ -169,6 +165,8 @@ func setupInstallCommand() *cli.Command {
 }
 
 func runInstallNodeSubCommand(ctx context.Context, config *configs.Config) (err error) {
+	//print(config.String())
+	//return nil
```
Comment (georgkcode0) [inline cmd/install.go:169]: should remove

File: configs/config.go [hunk 2]
```diff
@@ -2,71 +2,88 @@ package configs
...
-	// WalletLocalNetConfig - localnet config for walletnode
-	WalletLocalNetConfig = `pastel-api:
-  hostname: "127.0.0.1"
-  port: 29932
-  username: ""
-  password: ""
-  
+	// WalletDefaultConfig - default config for walletnode
+	WalletDefaultConfig = `
+pastel-api:
```
Comment (georgkcode0) [inline configs/config.go:17]: Better if we use `template` for easy understanding

File: cmd/start.go [hunk 3]
```diff
@@ -331,7 +331,7 @@ func runMasterNodOnColdHot(ctx context.Context, config *configs.Config) error {
 	log.WithContext(ctx).Info("Finished checking pastel config!")
 
 	// If create master node using HOT/HOT wallet
-	if pastelid, err = getMasternodeConf(ctx, config); err != nil {
+	if /*pastelid*/_, err = getMasternodeConf(ctx, config); err != nil {
```
Comment (georgkcode0) [inline cmd/start.go:334]: should remove

File: cmd/start.go [hunk 4]
```diff
@@ -382,7 +382,7 @@ func runMasterNodOnColdHot(ctx context.Context, config *configs.Config) error {
 
 	// ***************  8. Start supernode  **************
 
-	err = runSuperNodeRemote(ctx, config, client, extIP, pastelid)
+	err = runSuperNodeRemote(ctx, config, client/*, extIP, pastelid*/)
```
Comment (georgkcode0) [inline cmd/start.go:385]: should remove

File: cmd/start.go [hunk 5]
```diff
@@ -309,7 +309,7 @@ func runMasterNodOnHotHot(ctx context.Context, config *configs.Config) error {
 }
 
 func runMasterNodOnColdHot(ctx context.Context, config *configs.Config) error {
-	var pastelid string
+	//var pastelid string
```
Comment (georgkcode0) [inline cmd/start.go:312]: remove

File: configs/init.go [hunk 6]
```diff
@@ -13,7 +16,8 @@ type Init struct {
 	RemotePastelUtilityDir string `json:"remotepastelutilitydir,omitempty"`
 }
 
+/*
```
Comment (georgkcode0) [inline configs/init.go:19]: remove

File: cmd/install.go [hunk 7]
```diff
@@ -118,15 +120,10 @@ func setupSubCommand(config *configs.Config,
 		subCommand.SetActionFunc(func(ctx context.Context, args []string) error {
 			ctx, err := configureLogging(ctx, commandMessage, config)
 			if err != nil {
+				//Logger doesn't exist
 				return err
```
Comment (pjenys0) [inline cmd/install.go:124]: prefer to wrap error with context to let user easy to tracking.
like:
  `return errors.Wrap(err, "failed to configure logging option")`

File: cmd/install.go [hunk 8]
```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
...
-		log.WithContext(ctx).Info("The dd-service Installed Successfully")
 	}
 
 	return nil
 }
 
+func createInstallDir(ctx context.Context, config *configs.Config, installPath string) error {
+	defer log.WithContext(ctx).Infof("Install path is %s", installPath)
+
+	if err := utils.CreateFolder(ctx, installPath, config.Force); os.IsExist(err) {
```
Comment (pjenys0) [inline cmd/install.go:332]: missing the case utils.CreateFolder() failed and error is not satisfied os.IsExist(err).

File: cmd/install.go [hunk 9]
```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
...
+	return nil
+}
+
+func installExecutable(ctx context.Context, config *configs.Config,
+	downloadURL string,
+	archiveName string,
+	toolType constants.ToolType) (err error) {
+
+	if err = utils.DownloadFile(ctx, filepath.Join(config.PastelExecDir, archiveName), downloadURL); err != nil {
+		log.WithContext(ctx).Errorf(fmt.Sprintf("Failed to download pastel executable file : %s", downloadURL))
```
Comment (pjenys0) [inline cmd/install.go:401]: could you change:
`log.WithContext(ctx).Errorf(fmt.Sprintf("Failed to download pastel executable file : %s", downloadURL))`
to 
`log.WithContext(ctx).WithError(err).Errorf("Failed to download pastel executable file : %s", downloadURL)`

File: cmd/install.go [hunk 10]
```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
...
 
-func uncompressArchive(ctx context.Context, dstFolder string, archiveFile string) error {
-	file, err := os.Open(archiveFile)
-
-	if err != nil {
-		log.WithContext(ctx).Error("Not found archive file!!!")
-		return err
+func makeExecutable(ctx context.Context, dirPath string, fileName string) error {
+	if utils.GetOS() == constants.Linux {
+		if _, err := RunCMD("chmod", "777", filepath.Join(dirPath, fileName)); err != nil {
```
Comment (pjenys0) [inline cmd/install.go:482]: nit: should not set full permission for any group/any user.

File: cmd/install.go (hunk 9 above)
Comment (a-ok123) [reply cmd/install.go:401]: yes, missed this on

File: cmd/start.go (hunk 3 above)
Comment (a-ok123) [reply cmd/start.go:334]: this and related to it commented out temporary - i didn't review the code in start.go yet

File: cmd/install.go [hunk 11]
```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
...
 	}
-	return nil
-}
 
-// InitializeFunc - Initialize the function
-func InitializeFunc(ctx context.Context, config *configs.Config) (err error) {
-	configJSON, err := config.String()
+	fileName, err := utils.CreateFile(ctx, filepath.Join(workDirPath, configFileName), config.Force)
 	if err != nil {
+		log.WithContext(ctx).Errorf("Failed to create %s file", configFileName)
```
Comment (pjenys0) [inline cmd/install.go:503]: nit: should include error in same log, than log error in caller function - easy to associate error with command than.
or don't need to log here. simpy wrap error with context, like:
`return errors.Errorf("failed to create % file : %v", configFileName, err)`

File: cmd/install.go [hunk 12]
```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
...
 		return err
 	}
 
-	if err = config.SaveConfig(); err != nil {
+	if err = utils.WriteFile(fileName, toolConfig); err != nil {
+		log.WithContext(ctx).Errorf("Failed to write config to %s file", configFileName)
```
Comment (pjenys0) [inline cmd/install.go:508]: nit: should include error in same log, than log error in caller function - easy to associate error with command than.
or don't need to log here. simpy wrap error with context, like:
return errors.Errorf("failed to write config to % file : %v", configFileName, err)

File: cmd/install.go [hunk 13]
```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
...
+	config.RPCUser = utils.GenerateRandomString(8)
+	config.RPCPwd = utils.GenerateRandomString(15)
 
-		if _, err = RunCMD("chmod", "777",
-			filepath.Join(config.PastelExecDir, constants.ChromeExecFileName[utils.GetOS()])); err != nil {
-			log.WithContext(ctx).Error("Failed to make chrome-install as executable")
-			return err
-		}
+	// create pastel.conf file
+	f, err := utils.CreateFile(ctx, config.WorkingDir+"/pastel.conf", config.Force)
```
Comment (pjenys0) [inline cmd/install.go:533]: should use filepath.Join(config.WorkingDir, "pastel.conf")

File: cmd/install.go [hunk 14]
```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
...
-func installPackages(ctx context.Context) (err error) {
-	if utils.GetOS() == constants.Linux {
-		installedCmd := utils.GetInstalledCommand(ctx)
-		var notInstall []string
-		for _, p := range constants.DependenciesPackages {
-			if _, ok := installedCmd[p]; !ok {
-				notInstall = append(notInstall, p)
-			}
-		}
+func updatePastelConfigFile(ctx context.Context, fileName string, config *configs.Config) error {
```
Comment (pjenys0) [inline cmd/install.go:561]: nit: it's better to use  bytes.Buffer to join lines together before write all to file.

**Review by pjenys0 — APPROVED** [review]
Thanks.

**Pull request merged**